edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Test/ModuleIdentityLifecycleManagerTest.cs

```diff
@@ -237,5 +237,48 @@ public async Task TestGetModulesIdentity_WithUnchanged_ShouldReturnAllWhenReques
 
             Mock.Get(identityManager).Verify(im => im.DeleteIdentityAsync(Module3));
         }
+
+        [Fact]
+        [Unit]
+        public async Task TestGetModulesIdentity_WithBadModules_ShouldRemoveThoseIdentities()
+        {
+            // Arrange
+            const string Module1 = "module1";
+            var identity1 = new Identity(Module1, Guid.NewGuid().ToString(), "Me");
+
+            const string Module2 = "module2";
+            var identity2 = new Identity(Module2, Guid.NewGuid().ToString(), Constants.ModuleIdentityEdgeManagedByValue);
+
+            const string Module3 = "module3";
+            var identity3 = new Identity(Module3, Guid.NewGuid().ToString(), Constants.ModuleIdentityEdgeManagedByValue);
+
+            const string Module4 = "module4";
+            var identity4 = new Identity(Module4, Guid.NewGuid().ToString(), "Me");
+
+            var edgeAgentIdentity = new Identity(Constants.EdgeAgentModuleIdentityName, Guid.NewGuid().ToString(), Constants.ModuleIdentityEdgeManagedByValue);
+
+            var edgeHubIdentity = new Identity(Constants.EdgeHubModuleIdentityName, Guid.NewGuid().ToString(), Constants.ModuleIdentityEdgeManagedByValue);
+
+            var identityManager = Mock.Of<IIdentityManager>(
+                m =>
+                    m.GetIdentities() == Task.FromResult(new List<Identity>() { identity1, identity2, identity3, identity4, edgeAgentIdentity, edgeHubIdentity }.AsEnumerable()) &&
+                    m.DeleteIdentityAsync(Module3) == Task.FromResult(identity3));
+
+            var moduleIdentityLifecycleManager = new ModuleIdentityLifecycleManager(identityManager, ModuleIdentityProviderServiceBuilder, EdgeletUri);
+            var envVar = new Dictionary<string, EnvVal>();
+            var currentModule1 = new TestModule(Module1, "v1", "test", ModuleStatus.Running, new TestConfig("image"), RestartPolicy.OnUnhealthy, ImagePullPolicy.OnCreate, Constants.DefaultStartupOrder, DefaultConfigurationInfo, envVar);
+            var currentModule2 = new TestModule(Module2, "v1", "test", ModuleStatus.Running, new TestConfig("image"), RestartPolicy.OnUnhealthy, ImagePullPolicy.OnCreate, Constants.DefaultStartupOrder, DefaultConfigurationInfo, envVar);
+            ModuleSet desired = ModuleSet.Create(new IModule[] { currentModule1, currentModule2 });
```

Comment:
can you also add a new module to desired, remove one from current 
also should test with an identity that exists which is not in current, but is in desired and should not remove. It seems the code doesn't check the desired, as Varun also commented, once you fix that then the test should work.